pkg/cloudprovider/cloud.go

```diff
@@ -115,6 +115,9 @@ type LoadBalancer interface {
 	// Implementations must treat the *v1.Service parameter as read-only and not modify it.
 	// Parameter 'clusterName' is the name of the cluster as presented to kube-controller-manager
 	EnsureLoadBalancerDeleted(ctx context.Context, clusterName string, service *v1.Service) error
+	// GetLoadBalancerName returns the name of the load balancer. Implementations must treat the
+	// *v1.Service parameter as read-only and not modify it.
+	GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string
```

Comment:
I think this method should be placed below `GetLoadBalancer` since they're tightly coupled, thoughts?

Reply:
Yeah, I agree, that makes more sense.